addons/promote/common/src/main/java/org/commonjava/indy/promote/util/Batcher.java

```diff
@@ -0,0 +1,30 @@
+package org.commonjava.indy.promote.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class Batcher
+{
+    public static <T> Collection<Collection<T>> batch( Collection<T> collection, int batchSize )
+    {
+        Collection<Collection<T>> batches = new ArrayList<>();
```

Comment:
you can directly declare batches as ArrayList<Collection<T>>, so no need to do further casting again. And same to batch.